src/AzureAuth/Commands/Info/CommandInfoResetDeviceID.cs

```diff
@@ -15,6 +15,8 @@ namespace Microsoft.Authentication.AzureAuth.Commands.Info
     [Command(Name = "reset-device-id", Description = "Reset your AzureAuth telemetry device identifier.")]
     public class CommandResetDeviceID
     {
+        private const string ApplicationName = "azureauth";

```

Comment:
This creates duplication of the App name that we don't want, if the name ever needs to change we should be able to do that in 1 place. 

Instead, I think we can depend on (add the to argument list) the original `CommandLineApplication` that Lasso registers, and use the Name from that (this is how Lasso derives the app name internally).

Reply:
If I inject `CommandLineApplication`, I will get `CommandLineApplication<CommandResetDeviceID>` and its name is `reset-device-id`. I cannot inject `CommandLineApplication<CommandLineAzureAuth>`

Reply:
After looking at this in Lasso I was reminded that we don't need to do anything - we can always get the top level command by going through the current command's `Parent` property. So here, we can follow the `Parent` to get the top level command name. We could hardcode one reference to `.Parent` since we know there's only 1. It might be nice to write an integration test to pin this logic, so if the command ever moves, we know it still gets the right commend name.